implementations/cats/src/main/scala/sttp/client3/impl/cats/CatsMonadAsyncError.scala

```diff
@@ -17,13 +17,11 @@ class CatsMonadAsyncError[F[_]](implicit F: Concurrent[F]) extends MonadAsyncErr
     F.recoverWith(rt)(h)
 
   override def async[T](register: ((Either[Throwable, T]) => Unit) => Canceler): F[T] =
-    F.cancelable(register.andThen(c => F.delay(c.cancel)))
+    F.async_(register.andThen(c => F.delay(c.cancel)))
```

Comment:
I think you are actually losing the canceller here (and value discard makes up for it)

Instead:
```
F.async(cb => F.delay(register(cb)).map(c => F.delay(c.cancel).some))
```
Note the signatures:
```
  def async[A](k: (Either[Throwable, A] => Unit) => IO[Option[IO[Unit]]]): IO[A] 

  def async_[A](k: (Either[Throwable, A] => Unit) => Unit): IO[A]
```

Reply:
https://github.com/softwaremill/sttp/pull/738/commits/7ce2faaec9d0d9ecea7f7e9d28649b579062674d